src/cmd/singularity/cli/singularity.go

```diff
@@ -38,6 +50,21 @@ func init() {
 	SingularityCmd.Flags().BoolVarP(&quiet, "quiet", "q", false, "Suppress all normal output")
 	SingularityCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Increase verbosity +1")
 
+	usr, err := user.Current()
+	if err != nil {
+		sylog.Fatalf("Couldn't determine user home directory: %v", err)
+	}
+	defaultTokenFile := path.Join(usr.HomeDir, ".singularity", "sylabs-token")
+	// authToken priority default_file < env < file_flag
+	SingularityCmd.Flags().StringVar(&tokenFile, "tokenfile", defaultTokenFile, "path to the file holding your sylabs authentication token")
+	if val := os.Getenv("SYLABS_TOKEN"); val != "" {
+		authToken = val
+	}
+	if i := strings.Compare(tokenFile, defaultTokenFile); i != 0 {
+		authToken, authWarning = auth.ReadToken(tokenFile)
```

Comment:
I can't see where anything is done with the authWarning?

I don't think it should really be handled here. Would it be better for something like:

- If there is a gross problem with reading an explicit token file then it's a fatal here.
- If no token exists the warning gets handled inside the builder / library code when that code sees authToken == "" ?